File: src/examples/snippets.java

//! [Using a builder]
FlowElement element = new ElementBuilder(loggerFactory)
    .setOption(value)
    .setAnotherOption(anotherValue)
    .build(compulsoryOption);
//! [Using a builder]

//! [Using a pipeline builder]
Pipeline pipeline = new PipelineBuilder(loggerFactory)
    .setAutoDisposeElements(true)
    .setSuppressProcessException(true)
    .addFlowElement(element)
    .build();
//! [Using a pipeline builder]

//! [Get or add]
final TData aspectData = flowData.getOrAdd(getTypedDataKey(), getDataFactory());
//! [Get or add]

//! [Build in sequence]
Pipeline pipeline = new PipelineBuilder(loggerFactory)
    .addFlowElement(E1)
    .addFlowElement(E2)
    .build();
//! [Build in sequence]

//! [Build in parallel]
Pipeline pipeline = new PipelineBuilder(loggerFactory)
    .addFlowElementsParallel(new FlowElement[]{ E1, E2 })
    .build();
//! [Build in parallel]

//! [Iterate properties]
for (Map.Entry<String, Map<String, ElementPropertyMetaData>> elementProperties
    : pipeline.getElementAvailableProperties().entrySet()) {
    
    String elementName = elementProperties.getKey();
    
    for (Map.Entry<String, ElementPropertyMetaData> item
        : elementProperties.getValue().entrySet()) {
        
        String propertyName = item.getKey();
        ElementPropertyMetaData property = item.getValue();
    }
}
//! [Iterate properties]

File: src/examples/snippets.cs
using System;
using namespace FiftyOne.Pipeline.Core.FlowElements;

//! [Using a builder]
IFlowElement element = new ElementBuilder(loggerFactory)
    .SetOption(value)
    .SetAnotherOption(anotherValue)
    .Build(compulsoryOption);
//! [Using a builder]

//! [Get or add]
var elementData = flowData.GetOrAdd(ElementDataKeyTyped, CreateElementData);
//! [Get or add]

//! [Build in sequence]
IPipeline pipeline = new PipelineBuilder(loggerFactory)
    .AddFlowElement(E1)
    .AddFlowElement(E2)
    .Build();
//! [Build in sequence]

//! [Build in parallel]
IPipeline pipeline = new PipelineBuilder(loggerFactory)
    .AddFlowElementsParallel(E1, E2)
    .Build();
//! [Build in parallel]

//! [Iterate properties]
foreach (var elementProperties in pipeline.ElementAvailableProperties)
{
    var elementName = elementProperties.Key;
    
    foreach (var item in elementProperties.Value)
    {
        var propertyName = item.Key;
        var property = item.Value;
    }
}
//! [Iterate properties]